src/main/java/servlets/Calculator.java

```diff
@@ -0,0 +1,138 @@
+package servlets;
+
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+
+
+@WebServlet(
+        name = "Calculator",
+        urlPatterns = {"/calc"}
+)
+public class Calculator extends HttpServlet {
+    Stack<String>  stack = new Stack<>();
+    Stack<String>  nstack = new Stack<>();
+    int prev = 0;
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        PrintWriter out = resp.getWriter();
+        String equation = req.getParameter("equation");
+        Map<String, String[]> allMap = req.getParameterMap();
+        int a = 0;char[] modequation;
+        while (a!=2) {
+            modequation = equation.toCharArray();
+            for (int i = 0; i < modequation.length; i++)
+                if (Character.isLetter(modequation[i])){
+                  equation = equation.replaceAll(String.valueOf(modequation[i]),
+                          String.valueOf(allMap.get(String.valueOf(modequation[i]))[0]));
+                }
+            a++;
+        }
+        modequation = equation.toCharArray();
+        for (int i = 0; i < modequation.length; i++)
+            if ('0' <= modequation[i] && modequation[i] <= '9') {
+                int j = 0;
+                if (!stack.isEmpty() && prev == 1) {
+                    if (stack.peek().matches("[-+]?\\d+")) {
+                        j = Integer.parseInt(stack.peek().trim());
+                        stack.pop();
+                        stack.push(j + String.valueOf(modequation[i]));
+                        prev = 1;
+                    }
+                } else {
+                    stack.push(String.valueOf(modequation[i]));
+                    prev = 1;
+                }
+            } else {
+                if (modequation[i] == '+' || modequation[i] == '-') {
+                    if (!nstack.isEmpty()) {
+                        if (nstack.peek().equals("+") || nstack.peek().equals("-") ||
+                                nstack.peek().equals("*") || nstack.peek().equals("/")) {
+                            stack.push(String.valueOf(nstack.peek()));
+                            nstack.pop();
+                        }
+                    }
+                    nstack.push(String.valueOf(modequation[i]));
+                    prev = 0;
+                }
+                if (modequation[i] == '*' || modequation[i] == '/') {
+                    if (!nstack.isEmpty()) {
+                        if (nstack.peek().equals("*") || nstack.peek().equals("/")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4500113)